### DISCUSSION TITLE
Responsive Hero banner below main menu Hinode/Template

### DISCUSSION THREAD

**Discussion by Metisweb** [gethinode/hinode#493] [Help]
Dear All,

I am currently playing around with the Hinode/Template code in order to get a good grasp on it. For a couple websites I would need to be able to display a responsive Hero banner on certain pages.

Does the current code provide an option to display a responsive Hero banner below the main menu on certain pages?

If so, which piece of code should I use and where can I find it?
If not, what would be the best way to implement a responsive Hero banner on pages of choice?

Many thanks in advance.

Ton

**Comment by myrthos** [comment]
There is no hero banner functionality. However the part directly below the navigation bar, on the home page, is handled in `layouts/partials/home/featured.html`. So if you would add a hero banner yourself, I guess that would be the place to do it.

Hinode uses Bootstrap and the search for "Bootstrap hero banner" gives a number of examples on how to style a hero banner using bootstrap.

**Comment by Metisweb** [comment]
Thank you for your answer Myrthos.

The Hero banner would not have to be displayed on the home page but on a couple of single pages.

I guess I have to add some code at the top of the layouts/single.html file directly below line {{- define "partials/header.html" -}}
in order to include a Hero banner html block (in e.g. partials/xxx).

Am I on the right track ?

Many thanks

**Comment by myrthos** [comment]
If you want the hero banner to be as wide as the text (so having white space to the left and right) that is indeed one of the possible locations, with the actual hero code placed in a partial, as you mentioned. You can also put the hero code in a define in `single.html`, but I prefer to keep changes to Hinode code as small as possible and put the bulk in partials to prevent issues when merging with a newer version of Hinode.

However, I assume there is also a list page, which shows an overview of the available pages. If you want to use thumbnails to visualize those images, like [here](https://myrthos.net/blog/) you need to consider if you want to display that image on the same page that also has the hero banner, as that will happen by default. The image will then be displayed below the rest of the header and you end up with two wide images below each other (separated by the header). If you would like to use thumbnails in the list page, but not display them on the page, I would propose to define a not yet existing layout in the front matter of the page, like:

```yaml
layout: hero
```

And add "hero" to the slice in the line `{{ if not (in (slice "docs" "minimal") .Layout) }}`in the body define. In that case the thumbnail image is not being displayed on the page and you only have the hero banner.

if you don't use images for the list page, or you want them on the page, none of that is needed.

**Comment by Metisweb** [comment]
Basically what I am aiming at is to be able to display the thumbnail of a page straight under the navigation bar.
Maybe a simpler approach would be to implement this behaviour with some additional frontmatter settings. Something like the following.

thumbnail:
  position: top
  cover: full-width (full width of viewport, or default width as is right now)

Position 'top' would mean the thumbnail is displayed right under the navigation bar. All other stuff like the title, description, tags, post date etc. and social icons would be moved down under the thumbnail.
If the postion option is left blank everything stays the way it is right now.

Would this be a viable solution to explore further and could this be done relatively easy?

Many thanks.

**Comment by myrthos** [comment]
Ah I see. I thought it was about an actual hero image, with text on it and buttons.  
if it is about having the option of displaying the header below the image that might be something that Mark would be willing to add as an optional enhancement to Hinode. 
 
Currently the image handling is part of `body`. It could be taken out and placed in its own define. In `main` it could then be decided to show the image before `header` or after, based on a frontmatter setting. 

Displaying the image full width is quite a different change in the layout of Hinode. It can be done, by moving the display of the image before the navigation bars are displayed, but Mark has to decide if this is something he would like to support in Hinode.
It should be noted that currently Bootstrap limits the width of the largest container to be 1400 pixels. Actually when going larger than 1400 the container makes it even somewhat smaller than 1400 pixels. So 'full-width' is relative.

**Comment by markdumay** [comment]
Moving the thumbnail to the top within the current single page layout wouldn't be too difficult. However, in the current design, the thumbnail is constrained by the [single page's body](https://gethinode.com/docs/configuration/layout/#single-pages), set to .col-8. 

If I understand your request correctly, you would want to stretch the thumbnail to the full page width, e.g. `col-12`? Note that the page width is currently bound by [container-xxl](https://getbootstrap.com/docs/5.3/layout/containers/#how-they-work), which is `1320px`. This is to avoid layout issues on ultra-wide screens. And if you stretch it from `col-8` to `col-12`, you probably also want to adjust the thumbnail's aspect ratio? And is this on a page-by-page basis?

**Comment by Metisweb** [comment]
Hi Mark,

Yes, option to stretch to col-12. If this is bound to 1320px that is okay, because it is the max width of the website design, right?
Optional aspect ratio adjustment of thumbnail would be nice, indeed.
And, yes, preferably on a page-by-page basis for flexibility.

I really do think these adjustments to the design highly improve the usability of the template and will make it even more popular :)

Hugo and Hinode are relatively new to me at present, but once I get to grips with it I could imagine me contributing to the project...

Keep up the good work, Mark :)

**Comment by markdumay** [comment]
Alright, clear! Would you mind creating a feature request, so I can plan and track it?

**Comment by Metisweb** [comment]
Just did. Thx
